numpy/core/src/umath/ufunc_object.c

```diff
@@ -1243,7 +1243,7 @@ try_trivial_single_output_loop(PyArrayMethod_Context *context,
         int op_ndim = PyArray_NDIM(op[iop]);
 
         /* Special case 0-D since we can handle broadcasting using a 0-stride */
-        if (op_ndim == 0) {
+        if (op_ndim == 0 && iop != nin) {
```

Comment:
Should be `iop < nin`, I think, to ensure this doesn't bite ufuncs with more than one output (though I don't think there is any for which it is relevant; I don't think one can trigger the bug for single-input ufuncs, so `np.modf` seems safe)

Reply:
Yeah, the fast-path is only ever used for single outputs right now, but it is definitely the better spelling.

Reply:
Ah, that explains it, then!